// Copyright 2023 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Protobuffer for the generalized enriched cardinality sketch.
// SketchConfig is configuration of the sketch.
// Sketch is the sketch itself.

syntax = "proto3";

package wfa.frequency_count;

option java_package = "org.wfanet.frequencycount";
option java_multiple_files = true;
option java_outer_classname = "SecretShareProto";

message SecretShareParameter {
  uint32 modulus = 1;
}

// Seed to initialize the AES 256 counter mode.
message PrngSeed {
  bytes key = 1;
  bytes iv = 2;
}

message SecretShare {
  PrngSeed share_seed = 1;
  repeated uint32 share_vector = 2;
}
